src/key_info_managers/mod.rs

```diff
@@ -24,18 +27,21 @@ pub mod on_disk_manager;
 /// This structure corresponds to a unique identifier of the key. It is used internally by the Key
 /// ID manager to refer to a key.
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct KeyTriple {
-    app_name: ApplicationName,
-    provider_id: ProviderId,
+pub struct KeyIdentity {
+    /// The identity of the application that created the key.
+    application: ApplicationIdentity,
```

Comment:
Right, I see why creating `ApplicationIdentity` (and not flattening it) makes sense now 👌